docs/develop/go/test-suites.mdx

```diff
@@ -160,6 +161,109 @@ When an Activity sends a Heartbeat, be sure that you can see the Heartbeats in y
+that represents the operation without its implementation (basically, it represents the signature of
+the Nexus operation). You may also use the operation itself instead of creating the operation
+reference if you have it available.
+
+```go
+func (s *UnitTestSuite) Test_SimpleWorkflow_NexusSyncOperation() {
+        s.env.OnNexusOperation(
+                "sample-service",
+                nexus.NewOperationReference[SampleInput, SampleOutput]("sample-operation"),
+                SampleInput{},
+                workflow.NexusOperationOptions{},
+        ).Return(
+                &nexus.HandlerStartOperationResultSync[SampleOutput]{
+                        Value: SampleOutput{},
+                },
+                nil, // error if you want to simulate an error in the ExecuteOperation call
+        )
+        // You can also add a delay to return the mock values by calling After().
+        // Eg: s.env.OnNexusOperation(...).Return(...).After(1*time.Second)
+
+        s.env.ExecuteWorkflow(SimpleWorkflow, "test_nexus_operation")
+
+        s.True(s.env.IsWorkflowCompleted())
+        s.NoError(s.env.GetWorkflowError())
+}
+```
+
+Besides the synchronous operations, you can also mock asynchronous operations. The following example
+demonstrates how to test a Workflow executing a Nexus asynchronous operation. The returned value
+type in this case must be `*nexus.HandlerStartOperationResultAsync` with an `OperationID`, which can
+be any string of your choice. Furthermore, you must call `RegisterNexusAsyncOperationCompletion` to
+register the result of the asynchronous operation identified by the tuple service name, operation
+name, and operation ID.
+
+```go
+func (s *UnitTestSuite) Test_SimpleWorkflow_NexusAsyncOperation() {
+        s.env.OnNexusOperation(
+                "sample-service",
+                nexus.NewOperationReference[SampleInput, SampleOutput]("sample-operation"),
+                SampleInput{},
+                workflow.NexusOperationOptions{},
+        ).Return(
+                &nexus.HandlerStartOperationResultAsync{
+                        OperationID: "sample-operation-id",
+                },
+                nil, // error if you want to simulate an error in the ExecuteOperation call
+        )
+        err := env.RegisterNexusAsyncOperationCompletion(
+                "sample-service",
+                "sample-operation",
+                "sample-operation-id", // must match the OperationID above
+                SampleOutput{},
+                nil,            // error if you want to simulate an error in the operation
+                2*time.Second,  // delay to simulate how long the operation takes after it starts
+        )
+        s.NoError(err)
+        s.env.ExecuteWorkflow(SimpleWorkflow, "test_nexus_operation")
+
+        s.True(s.env.IsWorkflowCompleted())
+        s.NoError(s.env.GetWorkflowError())
+}
+```
+
+If your Workflow executed multiple Nexus asynchronous operations, you can mock each of them with
```

Comment:
Maybe add another section showing the the `WorkflowRunOperation` works with the test environment and explain that the client provided to `temporalnexus.NewSyncOperation` is fake and will panic if you try to use it.

Reply:
Why do we have the `temporalnexus.NewSyncOperation`?

Reply:
Checked in Slack, confirmed this issue is considered resolved.

Reply:
> Why do we have the temporalnexus.NewSyncOperation

Because it injects a client into the handler function, which we believe is useful.